### PR TITLE
fixed script syntax to access .keyword field, various cleanup

### PR DESCRIPTION
**Pull request by jpcarey**

### CODE REVIEW

File: docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc [hunk 1]
```diff
@@ -9,25 +9,23 @@ Example:
...
+        "combine_script" : "double profit = 0; for (t in params._agg.transactions) { profit += t } return profit",
+        "reduce_script" : "double profit = 0; for (a in params._aggs) { profit += a } return profit"
+      }
     }
+  }
 }
 --------------------------------------------------
 // CONSOLE
-// TEST[setup:ledger]
+// TEST[setup:transactions]
```
Comment (nik9000) [inline docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc]: The name of the setup script is `ledger` so this change broke the build. If you'd like to rename it you can change it in `docs/build.gradle` so long as you change everything that uses it.

**Review by nik9000 — CHANGES_REQUESTED** [review]

File: docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc (hunk 1 above)
Comment (jpcarey) [inline docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc]: reverted to the original `ledger` naming.

**Review by nik9000 — CHANGES_REQUESTED** [review]

File: docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc [hunk 2]
```diff
@@ -9,21 +9,19 @@ Example:
...
-                "combine_script" : "double profit = 0; for (t in params._agg.transactions) { profit += t } return profit",
-                "reduce_script" : "double profit = 0; for (a in params._aggs) { profit += a } return profit"
-            }
-        }
+  "size": 0,
+  "aggs": {
+    "profit": {
+      "scripted_metric": {
+        "init_script" : "params._agg.transactions = []",
+        "map_script" : "params._agg.transactions.add(doc['type.keyword'].value == 'sale' ? doc.amount.value : -1 * doc.amount.value)", <1>
```
Comment (jpcarey) [inline docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc]: makes sense. I was adding the `.keyword` as the default behavior from elasticsearch when no explicit mappings were provided. This page didn't have mappings for the examples (but the test does), so I assumed this was a miss.

**Comment by cbuescher** [comment]
@jpcarey are you still working on this? Going through some docs PRs to see if we can get some of these ready to get merged.

**Comment by cbuescher** [comment]
ping @jpcarey, do you need any more help with this? Just glancing at the PR, but it looks close to completion?

**Comment by colings86** [comment]
The documentation has changed quite a bit since this was opened and I'm not sure if this PR is still relevant. Since its not had any activity since the previous ping I'm going to close it but feel free to either re-open it with an up to date version or open a PR if there is still something you'd like to fix here

**Pull request closed without merging**